docs/docs/adrs/adr-015-partial-set-security.md

```diff
@@ -0,0 +1,293 @@
+
+In PSS, we distribute rewards only to validators that actually validated the consumer chain. To do this we generate a slice with bogus `VoteInfo` that contains the opted in (or Top N) validators and call [AllocateTokens](https://github.com/cosmos/cosmos-sdk/blob/v0.47.7/x/distribution/keeper/allocation.go#L14).
+
+```
+totalPower := 0
+var []abci.VoteInfo votes
+for _, val := range validators {
+    votes = append(votes, []abci.VoteInfo{{Validator: val, SignedLastBlock: true}})
+}
+
+k.AllocateTokens(ctx, totalPower, votes)
+```
+
+### Misbehaviour
+
+#### Fraud votes
+For fraud votes, we add a new type `FraudVoteProposal` similar to [EquivocationProposal](https://github.com/cosmos/interchain-security/pull/703). The time to detect incorrect execution evidence is 7 days (the same time as that of detecting a light client attack on a client with a `trustingPeriod` of 14 days); We have 7 days to detect an incorrect execution and then 14 days for the proposal to be accepted.
+
+##### When are fraud votes eligible?
+Because fraud votes allow the slashing of a validator through a governance proposal, we are cautious on when a fraud vote is eligible and try to restrict their eligibility as much as possible to avoid an adversary proposing a bogus fraud vote. For this, we only consider validators that have opted in in an Opt In chain. Fraud votes are also **only** intended to be used for **malicious** behaviour, for example, in a scenario similar to what [happened to Neutron](https://blog.neutron.org/neutron-halt-post-mortem-927dbe4540c8), validators should vote against slashing those validators.
+
+#### Message
+```
+message FraudVoteProposal {
+    string title = 1;
+    string description = 2;
+    
+    // validator that performed incorrect execution (consensus address on consumer chain)
+    Validator validator = 3;
+    
+    // validator's vote contains BlockId on what was signed
+    Vote vote = 4;
+    
+    // header for which the hash led to the blockId in the vote
+    Header header = 5;
+}
+```
+
+The message includes the `vote` that was signed by the validator. Validators that vote on a fraud vote can look at the `BlockID` contained in a `vote` and the `appHash`, `lastResultHash`, etc. in the `header` (that led to this `BlockID`) to conclude whether the to-be-voted validator has performed incorrect execution or not.
+
+#### Double signing
+We do not change the way slashing for double signing and light client attacks functions. If a validator misbehaves on a consumer, then we slash that validator on the provider.
+
+#### Downtime
```

Comment:
We will need to check that a validator is actually opted in to a consumer chain when receiving a downtime packet about them

Reply:
Good point. I clarified this here and added a note earlier that we need to keep track of validators that were opted in in the past even though they might not be opted in now.